internal/driver/glfw/menu_darwin.go

```diff
@@ -80,18 +81,22 @@ func addNativeMenu(w *window, menu *fyne.Menu, nextItemID int, prepend bool) int
 				C.bool(false),
 			)
 		} else {
-			C.insertDarwinMenuItem(
+			nsMenuItem := C.insertDarwinMenuItem(
 				nsMenu,
 				C.CString(item.Label),
 				C.int(nextItemID),
 				C.int(-1),
 				C.bool(item.IsSeparator),
 			)
+			if item.ChildMenu != nil {
+				nextItemID = addNativeSubMenu(w, nsMenuItem, item.ChildMenu, nextItemID)
+			}
 		}
 		if !item.IsSeparator {
-			action := item.Action // catch
-			callbacks = append(callbacks, func() { w.queueEvent(action) })
-			nextItemID++
+			if action := item.Action; action != nil { // catch action value
+				callbacks = append(callbacks, func() { w.queueEvent(action) })
+				nextItemID++
```

Comment:
I don't know for sure but it seems like nextItemID should not be subject to action != nil check?

Reply:
oh, yes … this happens when I get interrupted.